cmd/finch/support_bundle.go

```diff
@@ -0,0 +1,99 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/runfinch/finch/pkg/command"
+	"github.com/runfinch/finch/pkg/flog"
+	"github.com/runfinch/finch/pkg/lima"
+	"github.com/runfinch/finch/pkg/support"
+)
+
+func newSupportBundleCommand(logger flog.Logger, builder support.BundleBuilder, lcc command.LimaCmdCreator) *cobra.Command {
+	supportBundleCommand := &cobra.Command{
+		Use:   "support-bundle",
+		Short: "Gathers logs and info into a bundle to help debug issues",
+	}
+	supportBundleCommand.AddCommand(
+		newSupportBundleGenerateCommand(logger, builder, lcc),
+	)
+	return supportBundleCommand
+}
+
+func newSupportBundleGenerateCommand(logger flog.Logger, builder support.BundleBuilder, lcc command.LimaCmdCreator) *cobra.Command {
+	supportBundleGenerateCommand := &cobra.Command{
+		Use:   "generate",
+		Args:  cobra.NoArgs,
+		Short: "Generate support bundle",
+		RunE:  newGenerateSupportBundleAction(logger, builder, lcc).runAdapter,
+	}
+
+	supportBundleGenerateCommand.Flags().StringArray("include", []string{}, "additional files to include in the support bundle")
+	supportBundleGenerateCommand.Flags().StringArray("exclude", []string{}, "files to exclude from the support bundle")
+	return supportBundleGenerateCommand
+}
+
+type generateSupportBundleAction struct {
+	logger  flog.Logger
+	builder support.BundleBuilder
+	lcc     command.LimaCmdCreator
+}
+
+func newGenerateSupportBundleAction(
+	logger flog.Logger,
+	builder support.BundleBuilder,
+	lcc command.LimaCmdCreator,
+) *generateSupportBundleAction {
+	return &generateSupportBundleAction{
+		logger:  logger,
+		builder: builder,
+		lcc:     lcc,
+	}
+}
+
+func (a *generateSupportBundleAction) runAdapter(cmd *cobra.Command, args []string) error {
+	additionalFiles, err := cmd.Flags().GetStringArray("include")
+	if err != nil {
+		return err
+	}
+	excludeFiles, err := cmd.Flags().GetStringArray("exclude")
+	if err != nil {
+		return err
+	}
+	return a.run(additionalFiles, excludeFiles)
+}
+
+func (a *generateSupportBundleAction) run(additionalFiles []string, excludeFiles []string) error {
+	err := a.assertVMExists()
+	if err != nil {
+		return err
+	}
+	a.logger.Info("Generating support bundle...")
+	bundleFile, err := a.builder.GenerateSupportBundle(additionalFiles, excludeFiles)
+	if err != nil {
+		return err
+	}
+	a.logger.Infof("Bundle created: %s", bundleFile)
+	a.logger.Info("Files posted on a Github issue can be read by anyone.")
+	a.logger.Info("Please ensure there is no sensitive information in the bundle before uploading.")
```

Comment:
Should we list all the files (or the types of files) that will be included to be transparent?

Reply:
that's a good idea. I'll add a line for that